### PR TITLE
Adding windows CI workflow to JS

### PR DESCRIPTION
**Pull request by joshpalis**
Signed-off-by: Joshua Palis <jpalis@amazon.com>

Adding windows CI Workflow

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by bowenlan-amzn — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,13 +11,14 @@ jobs:
   build:
     strategy:
       matrix:
+        os: [ubuntu-latest, windows-latest]
         java: 
           - 8
           - 11
           - 14
 
     name: Build and Test
```
Comment (dbwiddis) [inline .github/workflows/ci.yml:20]: Consider `name: Build and Test JDK ${{ matrix.java }}, ${{ matrix.os }}`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -11,13 +11,14 @@ jobs:
   build:
     strategy:
       matrix:
+        os: [ubuntu-latest, windows-latest]
         java: 
           - 8
```
Comment (dbwiddis) [inline .github/workflows/ci.yml:16]: Consider putting the 3 `java` versions in a list the same way you did `os`, for consistency. `[8, 11, 14]`.  Is there a reason we aren't testing on 17?

**Review by dbwiddis — APPROVED** [review]
Approved, but see suggestions to make it better.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (joshpalis) [reply .github/workflows/ci.yml:20]: This is already reflected in the github checks. For example `Build and Test / Build and Test (ubuntu-latest, 11) `

File: .github/workflows/ci.yml (hunk 2 above)
Comment (joshpalis) [reply .github/workflows/ci.yml:16]: Sure I can modify this for java, as for why we aren't testing on 17 for 1.3 branch, I am not quite sure. However, java 11 and 17 are used to test for later versions of Job Scheduler

**Review by owaiskazi19 — APPROVED** [review]

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**